acvm-repo/acir_field/src/generic_ark.rs

```diff
@@ -446,6 +449,25 @@ mod tests {
             assert_eq!(minus_i_field_element.to_hex(), string);
         }
     }
+
+    #[test]
+    fn deserialize_even_and_odd_length_hex() {
+        // Test cases of (odd, even) length hex strings
+        let hex_strings =
+            vec![("0x0", "0x00"), ("0x1", "0x01"), ("0x002", "0x0002"), ("0x00003", "0x000003")];
+        for (i, case) in hex_strings.into_iter().enumerate() {
+            let i_field_element =
+                crate::generic_ark::FieldElement::<ark_bn254::Fr>::from(i as i128);
+            let odd_field_element =
+                crate::generic_ark::FieldElement::<ark_bn254::Fr>::from_hex(case.0).unwrap();
+            let even_field_element =
+                crate::generic_ark::FieldElement::<ark_bn254::Fr>::from_hex(case.1).unwrap();
+
+            assert_eq!(i_field_element, odd_field_element);
+            assert_eq!(odd_field_element, even_field_element);
+        }
+    }
```

Comment:
is there another place i should be testing the high level issue? to ensure that it works all the way up in top level `.nr` code

or is this sufficient?

Reply:
I think an example program in https://github.com/noir-lang/noir/tree/master/test_programs/execution_success would be good incase we switch from arkworks to another dependency for our field implementation

Reply:
aye aye 🫡 added in 1155e9f

just stuck it in the only existing test_program i saw with hex literals -- happy to break it out into its own if ya think the separation is warranted

Reply:
Looks great!